From 09892fb8d859166c3db6487aebb4e099dbe398ce Mon Sep 17 00:00:00 2001
From: Shengjiu Wang <shengjiu.wang@nxp.com>
Date: Thu, 12 Oct 2017 13:58:51 +0800
Subject: [PATCH 2626/5242] MLK-13946-2: hdp: add i2s clock for imx8qm hdmi
 audio

commit  12f76d18280994589bbff68f71e90a06c997e58a from
https://source.codeaurora.org/external/imx/linux-imx.git

hdmi audio need to enable the i2s clock and i2s_bypass clock

Signed-off-by: Shengjiu Wang <shengjiu.wang@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/gpu/drm/imx/hdp/imx-hdp.c |   21 ++++++++++++++++++++-
 drivers/gpu/drm/imx/hdp/imx-hdp.h |    2 ++
 2 files changed, 22 insertions(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/imx/hdp/imx-hdp.c b/drivers/gpu/drm/imx/hdp/imx-hdp.c
index 945a4bd..b3c18fa 100644
--- a/drivers/gpu/drm/imx/hdp/imx-hdp.c
+++ b/drivers/gpu/drm/imx/hdp/imx-hdp.c
@@ -367,7 +367,16 @@ static int dp_clock_init(struct imx_hdp *hdp)
 		dev_err(dev, "failed to get apb ctrl clk\n");
 		return PTR_ERR(hdp->clks.clk_apb_ctrl);
 	}
-
+	hdp->clks.clk_i2s = devm_clk_get(dev, "clk_i2s");
+	if (IS_ERR(hdp->clks.clk_i2s)) {
+		dev_err(dev, "failed to get i2s clk\n");
+		return PTR_ERR(hdp->clks.clk_i2s);
+	}
+	hdp->clks.clk_i2s_bypass = devm_clk_get(dev, "clk_i2s_bypass");
+	if (IS_ERR(hdp->clks.clk_i2s_bypass)) {
+		dev_err(dev, "failed to get i2s bypass clk\n");
+		return PTR_ERR(hdp->clks.clk_i2s_bypass);
+	}
 	return true;
 }
 
@@ -494,6 +503,16 @@ static int dp_ipg_clock_enable(struct imx_hdp *hdp)
 		dev_err(dev, "%s, pre clk apb ctrl error\n", __func__);
 		return ret;
 	}
+	ret = clk_prepare_enable(hdp->clks.clk_i2s);
+	if (ret < 0) {
+		dev_err(dev, "%s, pre clk i2s error\n", __func__);
+		return ret;
+	}
+	ret = clk_prepare_enable(hdp->clks.clk_i2s_bypass);
+	if (ret < 0) {
+		dev_err(dev, "%s, pre clk i2s bypass error\n", __func__);
+		return ret;
+	}
 	return ret;
 }
 
diff --git a/drivers/gpu/drm/imx/hdp/imx-hdp.h b/drivers/gpu/drm/imx/hdp/imx-hdp.h
index 1d876e5..23ecee7 100644
--- a/drivers/gpu/drm/imx/hdp/imx-hdp.h
+++ b/drivers/gpu/drm/imx/hdp/imx-hdp.h
@@ -146,6 +146,8 @@ struct hdp_clks {
 	struct clk *clk_apb_ctrl;
 	struct clk *av_pll_div;
 	struct clk *dig_pll_div;
+	struct clk *clk_i2s;
+	struct clk *clk_i2s_bypass;
 };
 
 struct imx_hdp {
-- 
1.7.9.5

